sdk/identity/test-resources-pre.ps1

```diff
@@ -0,0 +1,30 @@
+[CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+param (
+    # Captures any arguments from eng/New-TestResources.ps1 not declared here (no parameter errors).
+    [Parameter(ValueFromRemainingArguments = $true)]
+    $RemainingArguments
+)
+
+if ($EnvironmentVariables -eq $null) {
```

Comment:
does this work for local execution when .net defaults to using the protected outputs file rather than the environment?

Reply:
Yes this gets run within the scope of `New-TestResources.ps1` so `$EnvironmentVariables` is just a local variable, not actually the environment (which is `$env:*`).

Reply:
Right now it's a bit involved to get it set up until I can complete some of the helper scripts:

```
az keyvault secret download <base sub config> ...
az keyvault secret download <identity sub config> ...
$sc = get-content -raw <base> | ConvertFrom-Json -AsHashtable
$ic = get-content -raw <identity> | ConvertFrom-Json -AsHashtable
. ./eng/common/TestResources/SubConfig-Helpers.ps1
UpdateSubscriptionConfiguration $sc $ic | Out-Null
./eng/common/TestResources/New-TestResources.ps1 -ServiceDirectory identity -CI @sc
```

Which will eventually become something simpler like:

```
./eng/common/TestResources/New-TestResources.ps1 -ServiceDirectory identity -FromYamlSubscriptionConfig
```